Sprint 2/step 1

ArtemixxPG opened the pull request.

ArtemixxPG committed:
add custom hook useIngredients for context in burger-constructor

ArtemixxPG committed:
add POST for order
add dinamical counter sum order

andryxxa93 commented on react-burger/src/custom-hooks/use-ingredients.jsx:
```diff
@@ -0,0 +1,93 @@
+
+            dispatchSelectedIngredients({type: 'set', bun: bun, stuff: [...sauces, ...mains]})
+        }
+    }, [JSON.stringify(props.ingredients.buns), JSON.stringify(props.ingredients.sauces),
```
**Можно лучше**
Операции `JSON.stringify` довольно ресурсоёмкие и выходит, что пытаясь избежать лишних вызовов  `useEffect ` всё равно тратится много ресурсов. Рекомендую убрать `JSON.stringify`.

andryxxa93 commented on react-burger/src/custom-hooks/use-ingredients.jsx:
```diff
@@ -0,0 +1,93 @@
+export const IngredientsProvider = (props) => {
+
+
+    const [selectedIngredients, dispatchSelectedIngredients] = useReducer(reducer, selectedIngredient, undefined)
```
**Здорово**, что вы использовали `useReducer` - это полезный навык.

andryxxa93 commented on react-burger/src/custom-hooks/use-ingredients.jsx:
```diff
@@ -0,0 +1,93 @@
+    stuff: []
+}
+
+const IngredientContext = createContext(null)
```
**Необходимо исправить**
Контекст необходимо поместить внутрь директории `services` - она пригодится на в будущих спринтах когда будет внедрение `redux`.

andryxxa93 commented on react-burger/src/componets/burger-constructor/burger-constructor.jsx:
```diff
@@ -1,55 +1,79 @@
     })
 
+    useEffect(() => {
+        let result = summaryOrder(selectedIngredients)
```
**Необходимо исправить**
Там где не будет перезаписи переменной необходимо использовать `const`. Проверьте во всем проекте.

andryxxa93 commented on react-burger/src/componets/modal-content/modal-content-order-complete/modal-content-order-complete.jsx:
```diff
@@ -1,12 +1,13 @@
+import PropTypes from "prop-types";
 
-const ModalContentOrderComplete = () => {
+const ModalContentOrderComplete = (props) => {
```
**Можно улучшить**
Здесь и в коде других компонентов, лучше не писать каждый раз `props`, а использовать деструктуризацию, достав все необходимые данные из пропсов один раз. Это сделает код более понятным. https://learn.javascript.ru/destructuring#destrukturizatsiya-obekta

andryxxa93 reviewed (changes requested):
Хорошая работа!
Что понравилось:
- Хорошо разобрались с контекстом.
- Используете кастомные хуки.
- Нет лишнего кода.
- Классная анимация лого.

Что необходимо исправить:
- Контекст необходимо инициализировать внутри директории `services`.
- Не используйте `JSON.stringify` внутри зависимостей `useEffect` - это не принятая практика и больше съедает ресурсов чем бережёт.
- Используйте `const` вместо `let`.

Удачных правок!

ArtemixxPG committed:
add destructuring props
add folder 'services' for context
remove JSON.stringify from deps useEffects
use const from let

andryxxa93 reviewed (changes requested):
Проект стал теперь собираться с ошибками. Кое-где забыли убрать `props`.

<img width="1440" alt="Снимок экрана 2023-01-25 в 11 03 39" src="https://user-images.githubusercontent.com/60173362/214510364-f6f77943-3549-4234-a716-285a82fe63d8.png">

ArtemixxPG committed:
remove additional `props` from project

ArtemixxPG committed:
change calc unique key for constructor burger

andryxxa93 reviewed (approved):
Всё супер! Работа принята, удачного продолжения обучения!

ArtemixxPG committed:
change ror sprint-2/step-2
